### PR TITLE
Dont set --allow-privileged on kubernetes > 1.14 as it got removed

### PR DESCRIPTION
**Pull request by alvaroaleman**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

```release-note
none
```

/assign @nikhita

### CODE REVIEW

File: pkg/userdata/coreos/provider_test.go [hunk 1]
```diff
@@ -264,6 +264,29 @@ func TestUserDataGeneration(t *testing.T) {
 				DisableAutoUpdate: true,
 			},
 		},
+		{
+			name: "v1.15.0-vsphere",
+			providerSpec: &providerconfig.Config{
+				CloudProvider: "vsphere",
```
Comment (nikhita) [inline pkg/userdata/coreos/provider_test.go]: Why aren't SSH public keys needed here? :thinking:

Comment (alvaroaleman) [reply pkg/userdata/coreos/provider_test.go]: Well, they are not a mandatory property and I thought the fixture could be without a SSH key :)

**Pull request merged**